Services/Simulation/InternalInterpreter.cs

```diff
@@ -47,43 +50,43 @@ public InternalInterpreter(ILogger logger)
             this.random = new Random();
         }
 
-        public Dictionary<string, object> Invoke(
+        public void Invoke(
             string scriptPath,
             object scriptParams,
             Dictionary<string, object> context,
-            Dictionary<string, object> state)
+            ISmartDictionary state,
+            ISmartDictionary properties)
         {
             switch (scriptPath.ToLowerInvariant())
             {
                 case SCRIPT_RANDOM:
-                    return this.RunRandomNumberScript(scriptParams, state);
-
+                    this.RunRandomNumberScript(scriptParams, state);
+                    break;
                 case SCRIPT_INCREASING:
-                    return this.RunIncreasingScript(scriptParams, state);
-
+                    this.RunIncreasingScript(scriptParams, state);
+                    break;
                 case SCRIPT_DECREASING:
-                    return this.RunDecreasingScript(scriptParams, state);
-
+                    this.RunDecreasingScript(scriptParams, state);
+                    break;
                 default:
                     throw new NotSupportedException($"Unknown script `{scriptPath}`.");
             }
         }
 
         // For each sensors specified, generate a random number in the range requested
-        private Dictionary<string, object> RunRandomNumberScript(object scriptParams, Dictionary<string, object> state)
+        private void RunRandomNumberScript(object scriptParams, ISmartDictionary state)
         {
             var sensors = this.JsonParamAsDictionary(scriptParams);
             foreach (var sensor in sensors)
             {
                 (double min, double max) = this.GetMinMaxParameters(sensor.Value);
-                state[sensor.Key] = this.random.NextDouble() * (max - min) + min;
+                var value = this.random.NextDouble() * (max - min) + min;
```

Comment:
why var value = and not just set to the state directly like before?

Reply:
It was intended to match the style of the other methods below.